six/modules/c++/six.sicd/tests/test_add_additional_des.cpp

```diff
@@ -0,0 +1,252 @@
+    segmentWriter->attachSource(sSource);
+    writer.addAdditionalDES(segmentWriter);
+
+    std::string name(std::tmpnam(NULL));
+    try
+    {
+        writer.save(reinterpret_cast<const six::UByte*>(bandData.get()), name);
+        std::cout << "Test passed" << std::endl;
+        std::remove(name.c_str());
+        return true;
+    }
+    catch (except::Exception& e)
+    {
+        std::cerr << e.getMessage() << std::endl;
+        std::cerr << "Test failed" << std::endl;
+        return false;
+    }
+}
+
+bool canAddTwoSegmentWriters(const std::string& xmlName)
+{
+    std::cout << "Running canAddTwoSegmentWriters\n";
+    std::auto_ptr<six::Data> data = six::sicd::Utilities::readXML(xmlName);
+    mem::ScopedArray<sys::Int16_T> bandData(
+        generateBandData(*data));
+
+    six::Container container(six::DataType::COMPLEX);
+    container.addData(data.release());
+
+    six::NITFWriteControl writer;
+    writer.initialize(&container);
+
+    nitf::Record record = writer.getRecord();
+    nitf::DESegment desOne = record.newDataExtensionSegment();
+    desOne.getSubheader().getFilePartType().set("DE");
+    desOne.getSubheader().getTypeID().set("XML_DATA_CONTENT");
+    desOne.getSubheader().getVersion().set("01");
+    desOne.getSubheader().getSecurityClass().set("U");
+
+    static const char segmentOneData[] = "123456789ABCDEF0";
+    nitf::SegmentMemorySource sOneSource(segmentOneData, strlen(segmentOneData),
+        0, 0, true);
+    mem::SharedPtr<nitf::SegmentWriter> segmentOneWriter(new nitf::SegmentWriter);
+    segmentOneWriter->attachSource(sOneSource);
+    writer.addAdditionalDES(segmentOneWriter);
+
+    nitf::DESegment desTwo = record.newDataExtensionSegment();
+    desTwo.getSubheader().getFilePartType().set("DE");
+    desTwo.getSubheader().getTypeID().set("XML_DATA_CONTENT");
+    desTwo.getSubheader().getVersion().set("01");
+    desTwo.getSubheader().getSecurityClass().set("U");
+
+    static const char segmentTwoData[] = "123456789ABCDEF0";
+    nitf::SegmentMemorySource sTwoSource(segmentTwoData, strlen(segmentTwoData),
+        0, 0, true);
+    mem::SharedPtr<nitf::SegmentWriter> segmentTwoWriter(new nitf::SegmentWriter);
+    segmentTwoWriter->attachSource(sTwoSource);
+    writer.addAdditionalDES(segmentTwoWriter);
+
+    std::string name(std::tmpnam(NULL));
+    try
+    {
+        writer.save(reinterpret_cast<const six::UByte*>(bandData.get()), name);
+        std::cout << "Test passed" << std::endl;
+        std::remove(name.c_str());
+        return true;
+    }
+    catch (except::Exception& e)
+    {
+        std::cerr << e.getMessage() << std::endl;
+        std::cerr << "Test failed" << std::endl;
+        return false;
+    }
+}
+
+int main(int argc, char** argv)
+{
+    try
+    {
+        validateArguments(argc, argv);
+        std::string xmlName(argv[1]);
+
+        bool allPassed = true;
+        allPassed = addingNullSegmentWriterShouldThrow(xmlName) && allPassed;
```

Comment:
Can you easily add a test where you use the 5 and 283 length versions of [XML_DATA_CONTENT](https://github.com/mdaus/nitro/blob/master/modules/c/nitf/shared/XML_DATA_CONTENT.c)?  I'm not actually sure how you tell NITRO to do this, so if it's not straightforward, you can punt for now.